diag_manager/diag_manager.F90

```diff
@@ -3922,9 +3920,7 @@ SUBROUTINE diag_manager_init(diag_model_subset, time_init, err_msg)
     if (use_modern_diag) then
       CALL diag_yaml_object_init(diag_subset_output)
       CALL fms_diag_axis_object_init()
-      CALL fms_diag_object_init(255, 255) !< TO DO: MAX_LEN_VARNAME and MAX_LEN_META are supposed to be read from
-                                          !! the namelist and sent to fms_diag_object
-      fms_diag_files_object_initialized = fms_diag_files_object_init ()
+      CALL fms_diag_object%init(diag_subset_output) 
```

Comment:
This initializes the yaml twice 
`diag_yaml_object_init(diag_subset_output)` needs to be remove

Reply:
A module initialized was added to the yaml module so that it doesn';t crash.  The code needs to be rearranged so that the diag object handles the inialization of the yaml and the axis objects.